multi-agent/lib/mas/resources/builtin_service.py

```
@@ -0,0 +1,619 @@
+                    parent.ownership == ResourceOwnership.BUILTIN
+                    and parent.visibility == ResourceVisibility.PUBLIC
+                ):
+                    result.append(parent)
+                queue.append(parent_rid)
+        return result
+
+    def update_builtin_with_cascade(
+        self,
+        rid: str,
+        *,
+        config: dict = None,
```

Ruff (0.16.0)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)